sdk/loadtestservice/load-testing-rest/src/beginCreateOrUpdateTestRun.ts

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AbortController, AbortSignalLike } from "@azure/abort-controller";
+import { CancelOnProgress, OperationState, SimplePollerLike } from "@azure/core-lro";
+import { TestRunStatusPoller, PolledOperationOptions } from "./models";
+import { AzureLoadTestingClient, TestRunCreateOrUpdateParameters } from "./index.js";
+import { TestRunGet200Response } from "./responses";
+import { isUnexpected } from "./isUnexpected";
+import { sleep } from "./util/sleepLROUtility";
+
+/**
+ * Creates a poller to poll for test run status.
+ * @param client - The Load Testing client.
+ * @param options - The operation options.
+ * @returns A poller which can be called to poll until completion of the job.
+ */
+export async function beginCreateOrUpdateTestRun(
+  client: AzureLoadTestingClient,
+  testRunId: string,
+  testRunParams: TestRunCreateOrUpdateParameters,
+  polledOperationOptions: PolledOperationOptions = {}
+): Promise<TestRunStatusPoller> {
+  // Creating the test run
+  const testRunCreationResult = await client
+    .path("/test-runs/{testRunId}", testRunId)
+    .patch(testRunParams);
+
+  if (isUnexpected(testRunCreationResult)) {
+    throw testRunCreationResult.body.error;
+  }
```

Comment:
We generally want to avoid throwing in RLCs but this is an interesting situation. On one hand the operation failed to start but the function returns a poller, so can a poller convey to the user that the operation failed to start by seting the poller.getOperationState().status to failed and the error field to the actual error? I don't think that makes a whole lot of sense because there is no operation to poll for so returning a poller feels a bit misleading.

This situation is not encountered in other RLCs because the standard pollerHelper takes the initial response as input and it is the user responsibility to check if the initial response is valid/not-an-error before passing it to the helper.

This makes me think that we should pass the initial response to this function instead. @joheredi and @bterlson, what do you think?